### PR TITLE
Port examples to CMake and update old READMEs

### PR DESCRIPTION
**Pull request by ManuelJNunez**
:white_check_mark: I have updated the documentation accordingly.
:white_check_mark: I have read the CONTRIBUTING document.

### Summary

- Port flat_data_latency C++11 example to CMake.
- Update flat_data_latency C++11 README.md to the new template.
- Port flat_data_api C++98 example to CMake
- Update flat_data_api C++11 README.md to the new template.
- Update flat_data_api C++98 README.md to the new template.

### Details and comments

Ref issue #261 and issue #368

### CODE REVIEW

File: examples/connext_dds/flat_data_latency/c++11/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+include(ConnextDdsAddExample)
+
+connextdds_add_example(
+    IDL "CameraImage"
+    LANG "C++11"
+)
+
+target_link_libraries(flat_data_latency_publisher_cpp2
+    PRIVATE
+    RTIConnextDDS::metp
```
Comment (iblancasa) [inline examples/connext_dds/flat_data_latency/c++11/CMakeLists.txt:30]: Add one indentation level

File: examples/connext_dds/flat_data_latency/c++11/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+)
+
+target_link_libraries(flat_data_latency_subscriber_cpp2
+    PRIVATE
+    RTIConnextDDS::metp
```
Comment (iblancasa) [inline examples/connext_dds/flat_data_latency/c++11/CMakeLists.txt:35]: Add one indentation level

**Review by iblancasa — CHANGES_REQUESTED** [review]
Good job!

File: examples/connext_dds/flat_data_latency/c++11/CMakeLists.txt (hunk 1 above)
Comment (ManuelJNunez) [reply examples/connext_dds/flat_data_latency/c++11/CMakeLists.txt]: Done 👍🏻

File: examples/connext_dds/flat_data_latency/c++11/CMakeLists.txt (hunk 2 above)
Comment (ManuelJNunez) [reply examples/connext_dds/flat_data_latency/c++11/CMakeLists.txt]: Done 👍🏻

**Review by lulivi — APPROVED** [review]

**Review by fgaranda — APPROVED** [review]

**Review by iblancasa — APPROVED** [review]

**Pull request merged**